lib/casbin/model/policy.rb

```diff
@@ -1,12 +1,48 @@
 # frozen_string_literal: true
 
+require 'logger'
+
 module Casbin
   module Model
     class Policy
-      attr_reader :model
+      attr_reader :model, :logger
 
       def initialize
         @model = {}
+        @logger = Logger.new($stdout)
```

Comment:
move logger to attributs
see `lib/casbin/rbac/default_role_manager/role_manager.rb`

Reply:
We already have a reader for logger, do we also need a writer? Or you are talking about method parameters?

(And I think now it is not so important at all, since we will still need to think about a unified logging system in the library)

Reply:
about method parameters

Reply:
I think this is awkward to pass a logger to each constructor. Look:

We can create an enforcer with paths:
```ruby
e = Casbin::Enforcer.new 'path/to/model/config', 'path/to/policy/storage/file', custom_logger
```
or with already initialized model and adapter:
```ruby
model = Casbin::Model::Model.new custom_logger
# adapter = .......

# Then we should pass the same logger again
e = Casbin::Enforcer.new model, adapter, custom_logger
# or we can pass another logger, it is strange
# e = Casbin::Enforcer.new model, adapter, another_logger
```

Also we can specify `role_manager`, which again requires a logger:
```ruby
e.role_manager = Casbin::Rbac::DefaultRoleManager.new 5, custom_logger
```

And watchers may also need to specify the logger.

I propose something like this:

```ruby
Module Casbin
  class << self
    def logger
      @logger ||= Logger.new($stdout)
    end
    
    def logger=(value)
      @logger = value
    end
  end
end
```

and then

```ruby
Casbin.logger.info 'a log message'
```

so it will be like that:
```ruby
Casbin.logger = custom_logger

# all objects will use the same logger
e1 = Casbin::Enforcer.new 'path/to/model/config', 'path/to/policy/storage/file'

model = Casbin::Model::Model.new
adapter = SomeAdapter.new
e2 = Casbin::Enforcer.new model, adapter

e.role_manager = Casbin::Rbac::DefaultRoleManager.new 5
```

We can then use some null or spy logger in tests, use `Rails.logger` in Rails, etc.